src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -253,8 +253,8 @@ private static unsafe int Receive(SafeCloseSocket socket, SocketFlags flags, ILi
             }
             if (available == 0)
             {
-                // Always request at least one byte.
-                available = 1;
+                // Don't truncate iovecs.
+                available = int.MaxValue;
```

Comment:
@stephentoub ^- MSG_TRUNC turned out to be a corefx bug after all.

Reply:
Why is it ok to use available to truncate when available > 0 but not when available == 0?  Is that because for cases where truncation is a problem, all or none of the datagram will be in the buffer, and so if available > 0, we know it'll contain the full amount needed?

Reply:
Yes, when higher than zero, it's the size of the next pending datagram. If it's zero, there is a 0-size pending datagram or more likely: no datagram pending. In the latter case, the code would truncate to the first segment.